### DISCUSSION TITLE
Data handling after power outage

### DISCUSSION THREAD

**Discussion by Cube3SGM** [David00/rpi-power-monitor#114] [General]
Couple of questions with regards to how Power Monitor handles data and re-starting after a power outage.

1. When the monitor suffers a total loss of power (grid outage) and restarts on power up several hours later, how are the total consumption figures (Wh) dealt with.  The reason I ask is the graphs in Grafana show a linear line from last reading to start reading which would give and error on actual usage. I think it is only visual but thought I should check.
2. Easy one this one I think, On re-start from grid outage Grafana is working but it doesn't start collecting data until I sudu systemctl start power-monitor.service.  Could you give me some help with getting a un-attended re-start.

Thank you

3 Phase side question.  Can I monitor two 3 phase supplies.  I have 3 phase coming into the building, this is metered by the supply company for costs.  This supply is split into two distribution boards, 1 for offices and 1 for a workshop, as these are currently rented by two different companies. I would like to get an indication of consumption to these two boards. I don't need super accuracy, just an indication and a little insight to when the workshop and offices are being used.

Thank you
Regards Simon

**Comment by David00** [comment]
Hey @Cube3SGM!  Here's some feedback to your questions.

> 1. When the monitor suffers a total loss of power (grid outage) and restarts on power up several hours later, how are the total consumption figures (Wh) dealt with. The reason I ask is the graphs in Grafana show a linear line from last reading to start reading which would give and error on actual usage. I think it is only visual but thought I should check.

The total consumption (and production, for that matter) don't know about an outage. You're right that Grafana is just a visual thing where it connects to the last measured value and shows you a flat line. You can turn this off by editing the chart, and scrolling through the options in edit mode until you find the "Connect null values" option, and set it to Never.
![image](https://github.com/David00/rpi-power-monitor/assets/17997328/edd5bf05-3e7e-4082-80e2-09a743032b56)

> 2. Easy one this one I think, On re-start from grid outage Grafana is working but it doesn't start collecting data until I sudu systemctl start power-monitor.service. Could you give me some help with getting a un-attended re-start.

It is supposed to start up on boot automatically, but you can try to configure it again to do so with the command: `sudo systemctl enable power-monitor.service`. If that still doesn't work, try to go through the Creating the Service File steps again here in the docs:
https://david00.github.io/rpi-power-monitor/docs/v0.3.0/advanced-usage.html#creating-the-service-file


> 3 Phase side question. Can I monitor two 3 phase supplies. I have 3 phase coming into the building, this is metered by the supply company for costs. This supply is split into two distribution boards, 1 for offices and 1 for a workshop, as these are currently rented by two different companies. I would like to get an indication of consumption to these two boards. I don't need super accuracy, just an indication and a little insight to when the workshop and offices are being used.

This should be doable, but you'll need access to the main incoming feeds of each of the two distribution boards. It just depends on how the three phase is split up and delivered to the two distribution boards. You can monitor the amperage on any conductors - three phase doesn't get tricky until you want to also see the _power_ (as in, Watts) or _energy_ (Watt-hours) of the different phases.
